browser/components/newtab/MozContentSearch.js

```diff
@@ -239,6 +243,21 @@ MozContentSearch.prototype = {
     }.bind(this));
   },
 
+  _fireStateChangeEvent(engines) {
+    Task.spawn(function* () {
+      const allEngines = engines.map(
+        engineDetails => this._storeEngine(engineDetails)
+      );
+      const safeArray = new this._win.Array();
+      safeArray.push(...allEngines);
+      const eventInit = {
+        engines: safeArray
+      };
+      const event = new this._win.MozVisibleSearchEnginesChangeEvent("visibleenginechange", eventInit);
```

Comment:
Oh! @sarracini, it would be awesome if you could read over:
https://gist.github.com/marcoscaceres/13af01281730b5279ad6#implementing-event-emitting-object

I detail all this stuff there, but would be great to have someone sanity check what I wrote there (and maybe even propose improvements!)